spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/filter/NettyRoutingFilter.java

```diff
@@ -138,6 +139,11 @@ public int getOrder() {
 				String host = request.getHeaders().getFirst(HttpHeaders.HOST);
 				headers.add(HttpHeaders.HOST, host);
 			}
+
```

Comment:
The first one can be moved to a headers filter, however, the second one will need to get the native response in the `responseConnection` so that we can contribute to the `trailerHeaders` as per: https://github.com/reactor/reactor-netty/pull/1841/files#diff-370f6f6bd0d5120f944b9849362a341500700cc3d3ce0798634db78a20edd91cR484
Since for the request I was able to isolate it but not for the response since we need the `responseConnection`, I left both in the same filter.
What do you think?